ai_edge_torch/generative/layers/experimental/kv_cache.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2024 The AI Edge Torch Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# Cache management utilities.
+
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import torch
+import torch.utils._pytree as pytree
+
+from ai_edge_torch import hlfb
+from ai_edge_torch.generative.layers import model_config
+
+
+@dataclass
+class KVCacheEntry:
+  """A single cache entry include K and V caches.
+
+  Returns:
+      _type_: _description_
+  """
+
+  k_cache: torch.Tensor
+  v_cache: torch.Tensor
+
+  @classmethod
+  def from_model_config(
+      cls,
+      config: model_config.ModelConfig,
+      dtype: torch.dtype = torch.float32,
+      device: torch.device = None,
+  ) -> "KVCacheEntry":
+    shape = (
+        1,
+        config.kv_cache_max,
+        config.attn_config.num_query_groups,
+        config.head_dim,
+    )
+    k = torch.zeros(shape, dtype=dtype, device=device)
+    v = torch.zeros(shape, dtype=dtype, device=device)
+    obj = cls(k_cache=k, v_cache=v)
+    return obj
+
+
+@dataclass
+class KVCache:
```

Comment:
Would use name `ExpKVCache` or `EKVCache`. Something like that. It seems like a simple way to avoid confusion (or even name stomping) until we are ready to make the switchover.

Reply:
Done.